hashmap.js

```diff
@@ -25,7 +25,14 @@
 		this.clear();
 		switch (arguments.length) {
 			case 0: break;
-			case 1: this.copy(other); break;
+			case 1: {
+				if(Array.isArray(other)) {
+					this.from2DArray(other);

```

Comment:
Right, I missed the fact that `multi` works on flat arrays. You can flatten the incoming array with a single line. Can you try changing to this?
```js
var arr = Array.prototype.concat.apply([], other);
multi(this, arr);
```